### PR TITLE
Add docs for writing a custom oauth2 grant type

### PR DESCRIPTION
**Pull request by Rashmini**
## Purpose
$subject

## Related issue

- https://github.com/wso2/product-is/issues/18424

### CODE REVIEW

**Review by ashensw — APPROVED** [review]

File: en/identity-server/next/docs/references/extend/oauth2/write-a-custom-oauth-2.0-grant-type.md [hunk 1]
```diff
@@ -0,0 +1,341 @@
...
+    -   `            GrantTypeValidator           ` - This is used to
+        validate the grant request that is sent to the
+        `            /token           ` endpoint. You can define what
+        parameters must be in the request and define the validation of
+        them. You can write the new implementation by extending the 
+        `            AbstractValidator           `  in Apache Amber
+        component.
+2.  When implementation is done, package your class as a .jar file and
+    place it in the
+    `          <IS_HOME>/repository/component/lib         ` directory.
```
Comment (AnuradhaSK) [inline en/identity-server/next/docs/references/extend/oauth2/write-a-custom-oauth-2.0-grant-type.md:40]: shouldn't it need to be put into dropins?

File: en/identity-server/next/docs/references/extend/oauth2/write-a-custom-oauth-2.0-grant-type.md [hunk 2]
```diff
@@ -0,0 +1,341 @@
...
+!!! note
+    To generate the .jar file from the buildable source, run the following Apache Maven
+    command in the sample's location using the command line.
+
+    ``` java
+    mvn clean install
+    ```
+    
+2.  Copy the .jar file in target directory into the
+    `          <IS_HOME>/repository/component/lib         `
```
Comment (AnuradhaSK) [inline en/identity-server/next/docs/references/extend/oauth2/write-a-custom-oauth-2.0-grant-type.md:104]: same comment https://github.com/wso2/docs-is/pull/4268/files#r1432582017 applies here as well

File: en/identity-server/next/docs/references/extend/oauth2/write-a-custom-oauth-2.0-grant-type.md [hunk 3]
```diff
@@ -0,0 +1,341 @@
...
+-   [RoleBasedPasswordGrant](https://github.com/wso2/samples-is/blob/v4.5.6/oauth2/custom-grant/src/main/java/org/wso2/sample/identity/oauth2/grant/password/RoleBasedPasswordGrant.java) -
+    This does some RBAC validation apart from the authentication before
+    granting access.
+-   [ModifiedAccessTokenPasswordGrant](https://github.com/wso2/samples-is/blob/v4.5.6/oauth2/custom-grant/src/main/java/org/wso2/sample/identity/oauth2/grant/password/ModifiedAccessTokenPasswordGrant.java) -
+    This implementation customized the access token value.
+
+### Configure the resource
+
+1.  Copy the .jar file into the
+    `          <IS_HOME>/repository/component/lib         `
```
Comment (AnuradhaSK) [inline en/identity-server/next/docs/references/extend/oauth2/write-a-custom-oauth-2.0-grant-type.md:191]: should this be dropins?

File: en/identity-server/next/docs/references/extend/oauth2/write-a-custom-oauth-2.0-grant-type.md (hunk 1 above)
Comment (Rashmini) [reply en/identity-server/next/docs/references/extend/oauth2/write-a-custom-oauth-2.0-grant-type.md:40]: This doesn't work when the .jar file is put into dropins.

File: en/identity-server/next/docs/references/extend/oauth2/write-a-custom-oauth-2.0-grant-type.md (hunk 2 above)
Comment (Rashmini) [reply en/identity-server/next/docs/references/extend/oauth2/write-a-custom-oauth-2.0-grant-type.md:104]: Same as https://github.com/wso2/docs-is/pull/4268#discussion_r1433443588

File: en/identity-server/next/docs/references/extend/oauth2/write-a-custom-oauth-2.0-grant-type.md (hunk 3 above)
Comment (Rashmini) [reply en/identity-server/next/docs/references/extend/oauth2/write-a-custom-oauth-2.0-grant-type.md:191]: Same as https://github.com/wso2/docs-is/pull/4268#discussion_r1433443588

**Pull request merged**